examples/client.rs

```diff
@@ -0,0 +1,55 @@
+extern crate futures;
+extern crate hyper;
+extern crate tower;
+extern crate tower_buffer;
+extern crate tower_hyper;
+extern crate tower_in_flight_limit;
+extern crate tower_rate_limit;
+extern crate tower_reconnect;
+extern crate tower_retry;
+extern crate tower_service;
+extern crate tower_util;
+
+use futures::Future;
+use hyper::client::connect::Destination;
+use hyper::client::HttpConnector;
+use hyper::{Request, Response, Uri};
+use std::time::Duration;
+use tower::builder::ServiceBuilder;
+use tower_buffer::BufferLayer;
+use tower_hyper::client::{Builder, Connect};
+use tower_hyper::retry::{Body, RetryPolicy};
+use tower_hyper::util::Connector;
+use tower_in_flight_limit::InFlightLimitLayer;
+use tower_rate_limit::RateLimitLayer;
+use tower_reconnect::Reconnect;
+use tower_retry::RetryLayer;
+use tower_service::Service;
+
+fn main() {
+    hyper::rt::run(futures::lazy(|| request().map(|_| ())))
+}
+
+fn request() -> impl Future<Item = Response<hyper::Body>, Error = ()> {
+    let connector = Connector::new(HttpConnector::new(1));
+    let hyper = Connect::new(connector, Builder::new());
+
+    // let policy = RetryPolicy::new(5);
+    let dst = Destination::try_from_uri(Uri::from_static("http://127.0.0.1:3000")).unwrap();
+
+    let maker = ServiceBuilder::new(hyper)
+        // .add(RetryLayer::new(policy))
+        // .add(BufferLayer::new(5))
+        .add(InFlightLimitLayer::new(5))
+        .add(RateLimitLayer::new(5, Duration::from_secs(1)))
```

Comment:
Just checking, `add` means the new layer will wrap everything before it, right? So this ends up being `RateLimit<InFlightLimit<Hyper>>`, yea?

Reply:
Yes, I think that is correct. I need to fully test that, @carllerche would know more since he originally wrote the chain code for `tower-wb`

Reply:
would `.layer` be clearer?